src/Microsoft.FeatureManagement/Targeting/TargetingEvaluator.cs

```diff
@@ -36,29 +39,95 @@ public static bool IsTargeted(TargetingFilterSettings settings, ITargetingContex
 
             //
             // Check if the user is being targeted directly
+            if (settings.Audience.Users != null &&
+                IsTargeted(
+                    targetingContext,
+                    settings.Audience.Users,
+                    ignoreCase))
+            {
+                return true;
+            }
+
+            //
+            // Check if the user is in a group that is being targeted
+            if (settings.Audience.Groups != null &&
+                IsTargeted(
+                    targetingContext,
+                    settings.Audience.Groups,
+                    ignoreCase,
+                    hint))
+            {
+                return true;
+            }
+
+            //
+            // Check if the user is being targeted by a default rollout percentage
+            return IsTargeted(
+                targetingContext,
+                settings.Audience.DefaultRolloutPercentage,
+                ignoreCase,
+                hint);
+        }
+
+        /// <summary>
+        /// Determines if a targeting context is targeted by presence in a list of users
+        /// </summary>
+        public static bool IsTargeted(
+            ITargetingContext targetingContext,
+            IEnumerable<string> users,
+            bool ignoreCase)
+        {
+            if (targetingContext == null)
+            {
+                throw new ArgumentNullException(nameof(targetingContext));
+            }
+
+            if (users == null)
+            {
+                throw new ArgumentNullException(nameof(users));
+            }
+
             if (targetingContext.UserId != null &&
-                settings.Audience.Users != null &&
-                settings.Audience.Users.Any(user => targetingContext.UserId.Equals(user, GetComparisonType(ignoreCase))))
+                users.Any(user => targetingContext.UserId.Equals(user, GetComparisonType(ignoreCase))))
             {
                 return true;
             }
 
+            return false;
+        }
+
+        /// <summary>
+        /// Determine if a targeting context is targeted by presence in a group
+        /// </summary>
+        public static bool IsTargeted(
+            ITargetingContext targetingContext,
+            IEnumerable<GroupRollout> groups,
+            bool ignoreCase,
+            string hint)
```

Comment:
>string hint [](http://example.com/codeflow?start=12&length=11)

Should we add null check for `hint`?

Reply:
It's probably a good idea. But it looks like we don't enforce this at public entry point of feature manager. So to solve it right, I think I'll want to do a separate PR where I add enforcement through the call-chain.